mutex/src/test/workflows.test.ts

```diff
@@ -0,0 +1,108 @@
+import { TestWorkflowEnvironment } from '@temporalio/testing';
+import { WorkflowHandle } from '@temporalio/client';
+import { Runtime, DefaultLogger, Worker } from '@temporalio/worker';
+import { describe, before, after, it } from 'mocha';
+import { currentWorkflowIdQuery, hasLockQuery, lockWorkflow, testLockWorkflow } from '../workflows';
+import assert from 'assert';
+import { v4 as uuidv4 } from 'uuid';
+
+const taskQueue = 'test' + new Date().toLocaleDateString('en-US');
+
+describe('lock workflow', function () {
+  let runPromise: Promise<void>;
+  let worker: Worker;
+  let lockWorkflowHandle: WorkflowHandle<typeof lockWorkflow>;
+  let env: TestWorkflowEnvironment;
+
+  before(async function () {
+    this.timeout(10000);
+    Runtime.install({ logger: new DefaultLogger('WARN') });
+    env = await TestWorkflowEnvironment.create();
+
+    worker = await Worker.create({
+      connection: env.nativeConnection,
+      workflowsPath: require.resolve('../workflows'),
+      taskQueue,
+    });
+
+    runPromise = worker.run();
```

Comment:
nit: use a global bundle and worker per test with `runUntil`.
It's better for error handling an propagating fatal errors out of the worker.

Reply:
Fixed in https://github.com/temporalio/samples-typescript/pull/222/commits/86d3b3212a58c408e4d55919f18c0051b690eae8, please check it out :+1: